Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -422,5 +425,113 @@ private string GetFunctionSecretsFilePath(string functionName)
+        /// Populates a <see cref="ZipArchive"/> with the content of the function app.
+        /// It can also include local.settings.json and .csproj files for a full Visual Studio project.
+        /// sln file is not included since it changes between VS versions and VS can auto-generate it from the csproj.
+        /// All existing functions are added as content with "Always" copy to output.
+        /// </summary>
+        /// <param name="zip">the <see cref="ZipArchive"/> to be populated with function app content.</param>
+        /// <param name="includeAppSettings">Optional: indicates whether to add local.settings.json or not to the archive. Default is false.</param>
+        /// <param name="includeCsproj">Optional: indicates whether to add a .csproj to the archive. Default is false.</param>
+        /// <param name="projectName">Optional: the name for *.csproj file if <paramref name="includeCsproj"/> is true. Default is appName.</param>
+        /// <returns></returns>
+        public IList<ZipArchiveEntry> CreateArchive(ZipArchive zip, bool includeAppSettings = false, bool includeCsproj = false, string projectName = null)
+        {
+            var tracer = _traceFactory.GetTracer();
+            var directoryInfo = FileSystemHelpers.DirectoryInfoFromDirectoryName(_environment.FunctionsPath);
+
+            // First add the entire wwwroot folder at the root of the zip.
+            zip.AddDirectory(directoryInfo, tracer, string.Empty, out IList<ZipArchiveEntry> files);
+
+            if (includeAppSettings)
+            {
+                // include local.settings.json if needed.
+                files.Add(AddAppSettingsFile(zip));
+            }
+
+            if (includeCsproj)
+            {
+                // include .csproj for Visual Studio if needed.
+                projectName = projectName ?? ServerConfiguration.GetApplicationName();
+                files.Add(AddCsprojFile(zip, files, projectName));
+            }
+
+            return files;
+        }
+
+        /// <summary>
+        /// Creates a csproj file and adds it to the passed in ZipArchive
+        /// The csproj contain references to the core SDK, Http trigger, and build task
+        /// it also contain 'Always' copy entries for all the files that are in wwwroot.
+        /// </summary>
+        /// <param name="zip"><see cref="ZipArchive"/> to add csproj file to.</param>
+        /// <param name="files"><see cref="IEnumerable{ZipArchiveEntry}"/> of entries in the zip file to include in the csproj.</param>
+        /// <param name="projectName">the {projectName}.csproj</param>
+        private static ZipArchiveEntry AddCsprojFile(ZipArchive zip, IEnumerable<ZipArchiveEntry> files, string projectName)
+        {
+            const string microsoftAzureWebJobsVersion = "2.1.0-beta1";
+            const string microsoftAzureWebJobsExtensionsHttpVersion = "1.0.0-beta1";
+            const string microsoftNETSdkFunctionsVersion = "1.0.0-alpha4";
```

Comment:
that's true. I guess an alternative would be to query nuget.org for the latest. 

VS would work fine, it just might be using an outdated build task.